packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,94 @@
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import * as RadixLabel from "@radix-ui/react-label";
+import { useId } from "@reach/auto-id";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { Text } from "../text";
+import * as styles from "./Checkbox.css";
+import { IconChecked } from "./icons-svg/IconChecked";
+import { IconIndeterminate } from "./icons-svg/IconIndeterminate";
+import { IconUnchecked } from "./icons-svg/IconUnchecked";
+
+type CheckboxProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  {
+    checked?: "indeterminate" | boolean;
+    children: ReactNode;
+    defaultChecked?: "indeterminate" | boolean;
+    disabled?: boolean;
+    endDecorator?: ReactNode;
+    readonly?: boolean;
+  }
+>;
+
+export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
+  (
+    {
+      checked,
+      children,
+      className,
+      defaultChecked,
+      disabled,
+      endDecorator,
+      id: idProp,
+      readonly = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const id = useId(idProp);
+    const isDisabled = Boolean(disabled || readonly);
+
+    return (
+      <Flex
+        ref={ref}
+        {...styles.wrapper(
+          {
+            disabled: isDisabled,
+          },
+          className,
+        )}
+        {...props}
+      >
+        <Flex flexDirection="row" gap="xs">
+          <Box
+            asChild
+            {...styles.indicatorWrapper({
```

Comment:
I think, `root` makes more sense for this variable name as we are appending the style to root.

Reply:
`root` makes more sense to me for the `wrapper` case. But `wrapper` seems better among the two. 
We can rename it to `indicatorRoot` if it seems better. `root` maybe a bit misguiding.

Reply:
wrapper is more like a container, whereas root means the root of the component.

Reply:
it is a container imo :3

Reply:
Naming is hard for sure 👍 but I guess it shouldn't be indicatorWrapper for sure.